packages/lib/server/checkBookingLimits.ts

```diff
@@ -0,0 +1,89 @@
+import dayjs from "@calcom/dayjs";
+import prisma from "@calcom/prisma";
+import { BookingLimit } from "@calcom/types/Calendar";
+
+import { HttpError } from "../http-error";
+
+export async function checkBookingLimits(
+  bookingLimits: BookingLimit,
+  eventStartDate: Date,
+  eventId: number,
+  returnBusyTimes?: boolean
+) {
+  const limitCalculations = Object.entries(bookingLimits).map(
+    async ([key, limitingNumber]) =>
```

Comment:
I think this async, await is unnecessary as limitCalculations is expected to be an array of promises which can be achieved as 

```js
Object.entries(bookingLimits).map(
    ([key, limitingNumber]) =>
      checkLimit({ key, limitingNumber, eventStartDate, eventId, returnBusyTimes })
  );
```